pkg/repository/chartmuseum.go

```diff
@@ -99,7 +102,17 @@ type chartmuseum struct {
 func (c *chartmuseum) Start() {
 	c.logger.Info("Start to fetch")
 	_ = helm.RepoRemove(c.ctx, c.logger, c.repoName)
-	if err := helm.RepoAdd(c.ctx, c.logger, c.repoName, c.instance.Spec.URL, c.duration/2); err != nil {
+	if c.instance.Spec.AuthSecret != "" {
+		i := v1.Secret{}
+		if err := c.c.Get(c.ctx, types.NamespacedName{Name: c.instance.Spec.AuthSecret, Namespace: c.instance.GetNamespace()}, &i); err != nil {
+			c.logger.Error(err, "Failed to get secret")
+			return
+		}
+		c.username = string(i.Data["username"])
+		c.password = string(i.Data["password"])
```

Comment:
`username` and `password` have constant definitions here.
https://github.com/kubebb/core/blob/542ef80fde088f929800bcc15eb8ff7b695a4fb0/api/v1alpha1/repository.go#L26